src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/runtime/VMVersionMismatchException.java

```diff
@@ -27,25 +27,25 @@
 /** An instance of this exception is thrown when debuggee VM version
     is not supported current version of SA. */
 public class VMVersionMismatchException extends RuntimeException {
-    public VMVersionMismatchException(String supported, String target) {
+    public VMVersionMismatchException(Runtime.Version supported, Runtime.Version target) {
         super();
         supportedVersions = supported;
         targetVersion = target;
     }
 
     public String getMessage() {
-        return "Supported versions are " + supportedVersions +
-                ". Target VM is " + targetVersion;
+        return "Supported versions are " + supportedVersions.toString() +
```

Comment:
Why do you call to `toString()` explicitly here? String concatenation will do it for you without it.

Reply:
I'm not particular about `toString()` call, so I removed them.